sklearn/utils/_param_validation.py

```diff
@@ -43,10 +43,11 @@ def validate_parameter_constraints(parameter_constraints, params, caller_name):
     caller_name : str
         The name of the estimator or function or method that called this function.
     """
-    if params.keys() != parameter_constraints.keys():
+    if len(set(parameter_constraints.keys()) - set(params.keys())) != 0:
         raise ValueError(
-            f"The parameter constraints {list(parameter_constraints.keys())} do not "
-            f"match the parameters to validate {list(params.keys())}."
+            f"The parameter constraints {list(parameter_constraints.keys())}"
+            " contain unexpected parameters"
+            f" {set(parameter_constraints.keys())-set(params.keys())}"
         )
 
     for param_name, param_val in params.items():
```

Comment:
We need to add something like
```py
        # We allow parameters to not have a constraint so that third party estimators
        # can inherit from sklearn estimators without having to necessarily use the
        # validation tools. 
        if param_name not in parameter_constraints:
            continue
```

otherwise it will fail if there's a parameter without constraint.

Please also add a test in `test_param_validation.py` for that case: a function with less constraints than params